lib/rules/function-calc-no-unspaced-operator/__tests__/index.mjs

```diff
@@ -548,15 +566,139 @@ testRule({
+				{ message: messages.expectedBefore('-'), line: 1, endLine: 1, column: 26, endColumn: 27 },
+				{ message: messages.expectedAfter('-'), line: 1, endLine: 1, column: 26, endColumn: 27 },
+				{ message: messages.expectedBefore('-'), line: 1, endLine: 1, column: 30, endColumn: 31 },
+				{ message: messages.expectedAfter('-'), line: 1, endLine: 1, column: 30, endColumn: 31 },
+			],
+		},
+		{
+			code: 'a { top: calc((1px+ 1px)); }',
+			fixed: 'a { top: calc((1px + 1px)); }',
+			description: 'nested math expression',
+			message: messages.expectedBefore('+'),
+			line: 1,
+			column: 19,
+			endLine: 1,
+			endColumn: 20,
+		},
+		{
+			code: 'a { top: calc(calc(1px+ 1px)); }',
+			fixed: 'a { top: calc(calc(1px + 1px)); }',
+			description: 'nested math expression',
+			message: messages.expectedBefore('+'),
+			line: 1,
+			column: 23,
+			endLine: 1,
+			endColumn: 24,
+		},
+		{
+			code: 'a { top: calc(((1px+ 1px))); }',
+			fixed: 'a { top: calc(((1px + 1px))); }',
+			description: 'nested math expression',
+			message: messages.expectedBefore('+'),
+			line: 1,
+			column: 20,
+			endLine: 1,
+			endColumn: 21,
+		},
+		{
+			code: 'a { font-size: clamp(1rem, 2.5vw, 1rem+1rem); }',
+			fixed: 'a { font-size: clamp(1rem, 2.5vw, 1rem + 1rem); }',
+			description: 'multiple argument functions',
+			warnings: [
+				{ message: messages.expectedBefore('+'), line: 1, endLine: 1, column: 39, endColumn: 40 },
+				{ message: messages.expectedAfter('+'), line: 1, endLine: 1, column: 39, endColumn: 40 },
+			],
+		},
+		{
+			code: 'a { top: clamp(1px +2px, 3px-4px, none); }',
+			fixed: 'a { top: clamp(1px + 2px, 3px - 4px, none); }',
+			description: 'multiple argument functions',
+			warnings: [
+				{ message: messages.expectedAfter('+'), line: 1, endLine: 1, column: 20, endColumn: 21 },
+				{ message: messages.expectedBefore('-'), line: 1, endLine: 1, column: 29, endColumn: 30 },
+				{ message: messages.expectedAfter('-'), line: 1, endLine: 1, column: 29, endColumn: 30 },
+			],
+		},
+		{
+			code: 'a { width: min(25vw+25vw); }',
+			fixed: 'a { width: min(25vw + 25vw); }',
+			description: 'multiple argument functions',
+			warnings: [
+				{ message: messages.expectedBefore('+'), line: 1, endLine: 1, column: 20, endColumn: 21 },
+				{ message: messages.expectedAfter('+'), line: 1, endLine: 1, column: 20, endColumn: 21 },
+			],
+		},
+		{
+			code: 'a { padding: calc(1px+ 2px) calc(3px+ 4px); }',
+			fixed: 'a { padding: calc(1px + 2px) calc(3px + 4px); }',
+			description: 'nested math expression',
+			warnings: [
+				{ message: messages.expectedBefore('+'), line: 1, endLine: 1, column: 22, endColumn: 23 },
+				{ message: messages.expectedBefore('+'), line: 1, endLine: 1, column: 37, endColumn: 38 },
+			],
+		},
+		{
+			code: 'a { top: rgb(from red calc(r+1) calc(g-+2) calc(clamp(10px+2px, g+2, none))); }',
+			fixed:
+				'a { top: rgb(from red calc(r + 1) calc(g- + 2) calc(clamp(10px + 2px, g + 2, none))); }',
```

Comment:
questions
medium

What's `g-`?
Shouldn't it be fixed to `calc(g - +2)` instead?
i.e. `-` is subtraction and `+` is the unary operator

Reply:
`g-` is an ident token while we currently only scan for dimension tokens (e.g. `10px-`)

https://github.com/stylelint/stylelint/blob/4055b41b85ea9c1c1e2c529c5c4a97c7e289bc98/lib/rules/function-calc-no-unspaced-operator/index.mjs#L210

We can extend L210 to also look for idents and add extra logic to support both the `value` of an ident and the `unit` of a dimension.

Reply:
I think that @ybiquitous wouldn't mind if you were to commit on this branch.

Reply:
This problem is resolved thanks to PR #7676.